lib/3scale/backend/worker_async.rb

```diff
@@ -25,33 +27,28 @@ def initialize(options = {})
 
         @max_concurrent_jobs = configuration.async_worker.max_concurrent_jobs ||
             DEFAULT_MAX_CONCURRENT_JOBS
-
-        @reactor = Async::Reactor.new
       end
 
       def work
-        if one_off?
-          Async { process_one }
-          return
-        end
+        Sync do
+          if one_off?
+            process_one
+            return
```

Comment:
This line fails in tests, but also when run locally...
```
LocalJumpError: unexpected return
```

I'm not sure though what's the proper way to return from that method :thinking:

Reply:
Avoiding `return` as below apparently works, but I think there should be some other way... 
```
          if one_off?
            process_one
          else
            ...
            process_all
            ...

            unregister_worker
          end
```

Reply:
I refactored it to remove the global reactor and the barriers, and replace it all by different reactors for each step. This seems to have solved the problem.

Also, FYI @mayorova @akostadinov, I forgot to mention that all tests are actually executed twice: once in sync mode and once in async mode. See the output to find where each iteration starts:

- Sync: https://app.circleci.com/pipelines/github/3scale/apisonator/964/workflows/9dcf62e5-f121-4557-b83e-c89bba3605d0/jobs/1832/parallel-runs/0/steps/0-107?invite=true#step-107-97_41
- Async: https://app.circleci.com/pipelines/github/3scale/apisonator/964/workflows/9dcf62e5-f121-4557-b83e-c89bba3605d0/jobs/1832/parallel-runs/0/steps/0-107?invite=true#step-107-24703_42